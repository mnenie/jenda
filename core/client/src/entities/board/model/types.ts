import type { User } from '@/entities/user/model'

interface DateParams {
  createdAt?: Date
  updatedAt?: Date
  deletedAt?: Date
}

export interface Label {
  name: string
  color: string
}

export type Status = 'active' | 'archived'

type Priority = 'none' | 'low' | 'medium' | 'high'

export interface BoardRow extends Omit<Board, 'columns'> {
  tasks?: number
}

export interface Board extends DateParams {
  _id: string
  name: string
  columns?: Column[]
  status: Status
  labels: Label[]
  users: User[]
  color?: string
  estimate?: number
  // fix
  date?: string
}

interface Tag {
  _id: string
  name: string
}

export interface Card extends DateParams {
  _id: string
  title: string
  priority: Priority
  tags?: Tag[]
  chat?: boolean
  chatCount?: number
  users: User[]
}

export interface Column extends DateParams {
  _id: string
  title: string
  cards?: Card[]
}

export interface StatusBadge {
  _id: string
  indicator: string
  status: Status
}
