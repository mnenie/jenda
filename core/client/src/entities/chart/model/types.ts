export interface Chart {
  key: "users" | "boards" | "tasks" | "online";
  section: string;
}
