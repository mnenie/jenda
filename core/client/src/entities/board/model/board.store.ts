import { ref } from 'vue'
import { acceptHMRUpdate, defineStore } from 'pinia'
import type { Board, BoardRow } from './types'

export const useBoardsStore = defineStore('boards', () => {
  const boards = ref<BoardRow[]>([])
  const board = ref<Board>()

  function removeBoards(idxs: string[]) {
    idxs.forEach((id) => {
      const index = boards.value.findIndex(board => board._id === id)
      if (index !== -1) {
        boards.value.splice(index, 1)
      }
    })
  }

  function addBoard(board: Board) {
    boards.value.push(board)
  }

  return {
    boards,
    board,
    addBoard,
    removeBoards,
  }
})

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useBoardsStore, import.meta.hot))
}
