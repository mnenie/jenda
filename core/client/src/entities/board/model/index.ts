export * from './board.store'
export * from './composables'
export * from './types'
