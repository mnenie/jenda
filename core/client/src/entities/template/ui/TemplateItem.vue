<script setup lang="ts">
import type { Template } from '../model'
import { UiBadge } from '@/shared/ui'

defineProps<{
  template: Template
}>()
</script>

<template>
  <div
    class="relative flex flex-col justify-between items-start h-fit w-100% border border-solid border-neutral-200
    rounded-lg cursor-pointer dark:(bg-neutral-700/40 border-neutral-600)"
  >
    <div class="py-2 px-3">
      <UiBadge
        variant="outline"
        class="mb-3 !bg-blue-100 text-blue-500 dark:(!bg-blue-800 !text-blue-100 !border-blue-800)"
      >
        {{ template.tag }}
      </UiBadge>
      <div class="flex flex-col gap-1.5 mb-3">
        <p class="text-sm">
          {{ template.title }}
        </p>
        <span
          class="text-xs text-neutral-500 overflow-hidden text-ellipsis line-clamp-2 break-words
          dark:text-neutral-400"
        >
          {{ template.description }}
        </span>
      </div>
      <div class="flex items-center w-full justify-between">
        <span class="text-xs text-neutral-600 dark:text-neutral-300">{{ template.date }}</span>
        <img
          v-tooltip.bottom-end="$t('templates.user')"
          :src="template.user"
          class="w-5 h-5 rounded-full"
        />
      </div>
    </div>
  </div>
</template>
