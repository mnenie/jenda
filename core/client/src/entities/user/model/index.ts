export * from "./types";
export * from "./user.store";
