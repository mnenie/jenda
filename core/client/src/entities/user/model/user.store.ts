import type { User } from './types'
import { defineStore } from 'pinia'
import { ref } from 'vue'

export const useUserStore = defineStore('user', () => {
  const user = ref({} as User)

  return {
    user,
  }
})
