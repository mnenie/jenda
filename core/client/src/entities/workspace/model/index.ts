export * from './types'
export * from './workspace.store'
