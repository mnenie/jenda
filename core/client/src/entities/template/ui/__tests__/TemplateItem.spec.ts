import { shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import { templatesInfo } from '../../model'
import TemplateItem from '../TemplateItem.vue'
import i18n from '@/shared/libs/i18n'

describe('tests for TemplateItem.vue', () => {
  const wrapper = shallowMount(TemplateItem, {
    global: {
      plugins: [i18n],
      mocks: {
        t: (key: string) => {
          const translations: Record<string, string> = {
            'templates.user': 'test user',
          }
          return translations[key]
        },
      },
    },
    props: {
      template: templatesInfo[0],
    },
  })

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
