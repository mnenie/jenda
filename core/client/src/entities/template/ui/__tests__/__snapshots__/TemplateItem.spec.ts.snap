// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for TemplateItem.vue > should render correctly 1`] = `
"<div class="relative flex flex-col justify-between items-start h-fit w-100% border border-solid border-neutral-200 rounded-lg cursor-pointer dark:bg-neutral-700/40 dark:border-neutral-600">
  <div class="py-2 px-3">
    <ui-badge-stub class="mb-3 !bg-blue-100 text-blue-500 !dark:bg-blue-800 !dark:text-blue-100 !dark:border-blue-800" variant="outline"></ui-badge-stub>
    <div class="flex flex-col gap-1.5 mb-3">
      <p class="text-sm">Base Kanban</p><span class="text-xs text-neutral-500 overflow-hidden text-ellipsis line-clamp-2 break-words dark:text-neutral-400">Create a basic project with "Base Kanban" template</span>
    </div>
    <div class="flex items-center w-full justify-between"><span class="text-xs text-neutral-600 dark:text-neutral-300">June 12, 2024</span><img src="https://avatars.githubusercontent.com/u/121057011?v=4" class="w-5 h-5 rounded-full"></div>
  </div>
</div>"
`;
